src/Servers/Kestrel/Core/src/Internal/Infrastructure/Heartbeat.cs

```diff
@@ -27,58 +27,53 @@ public Heartbeat(IHeartbeatHandler[] callbacks, ISystemClock systemClock, IDebug
             _debugger = debugger;
             _trace = trace;
             _interval = Interval;
+            _timerThread = new Thread(state => ((Heartbeat)state).TimerLoop())
+            {
+                Name = "Kestrel Timer",
+                IsBackground = true
```

Comment:
In LibuvThread, we set IsBackground to false in debug so we can find issues where the thread isn't properly stopping.

Reply:
Sure but the libuv thread does more than this. I wouldn’t compare the 2. We could set it to true in debug if you’re paranoid about the timer not shutting down.

Reply:
How/when does this get initiated? Is it lazy and triggered by a first request? i.e. does it need executioncontext flow supression?

Reply:
Don't want an HttpContext stuck hanging around on a Thread 😉

Reply:
> How/when does this get initiated? Is it lazy and triggered by a first request? i.e. does it need executioncontext flow supression?

Startup. Could suppress execution context but that isn't new behavior. The Timer before didn't suppress it 😄 , nothing new here.